frontend/src/components/widgets/CameraInput/WebcamComponent.tsx

```diff
@@ -0,0 +1,175 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SwitchCamera } from "@emotion-icons/material-rounded"
+import { Video } from "@emotion-icons/open-iconic"
+import { useTheme } from "emotion-theming"
+import React, { ReactElement, useRef, useState } from "react"
+import { isMobile } from "react-device-detect"
+import Webcam from "react-webcam"
+
+import Button, { Kind } from "src/components/shared/Button"
+import Icon from "src/components/shared/Icon"
+import Tooltip, { Placement } from "src/components/shared/Tooltip"
+import { Theme } from "src/theme"
+import themeColors from "src/theme/baseTheme/themeColors"
+import CameraInputButton from "./CameraInputButton"
+import {
+  StyledBox,
+  StyledCameraInput,
+  StyledDescription,
+  StyledLink,
+  StyledSwitchFacingModeButton,
+} from "./styled-components"
+
+export interface Props {
+  handleCapture: (capturedPhoto: string | null) => void
+  width: number
+  disabled: boolean
+  clearPhotoInProgress: boolean
+  setClearPhotoInProgress: (clearPhotoInProgress: boolean) => void
+}
+
+export enum FacingMode {
+  USER = "user",
+  ENVIRONMENT = "environment",
+}
+
+export enum WebcamPermission {
+  PENDING = "pending",
+  SUCCESS = "success",
+  ERROR = "error",
+}
+
+interface AskForCameraPermissionProps {
+  width: number
+}
+
+export const AskForCameraPermission = ({
+  width,
+}: AskForCameraPermissionProps): ReactElement => {
+  return (
+    <StyledBox width={width}>
+      <Icon size="threeXL" color={themeColors.gray60} content={Video} />
+      <StyledDescription>
+        This app would like to use your camera.
+        <StyledLink href="https://streamlit.io">
+          Learn how to allow access.
+        </StyledLink>
+      </StyledDescription>
+    </StyledBox>
+  )
+}
+
+const WebcamComponent = ({
+  handleCapture,
+  width,
+  disabled,
+  clearPhotoInProgress: inClearPhotoMode,
+  setClearPhotoInProgress,
+}: Props): ReactElement => {
+  const [webcamPermission, setWebcamPermissionState] = useState(
+    WebcamPermission.PENDING
+  )
+  const videoRef = useRef<Webcam>(null)
+  const [facingMode, setFacingMode] = useState(FacingMode.USER)
+
+  function capture(): void {
+    if (videoRef.current !== null) {
+      const imageSrc = videoRef.current.getScreenshot()
+      handleCapture(imageSrc)
+    }
+  }
+
+  function switchCamera(): void {
+    setFacingMode(prevState =>
+      prevState === FacingMode.USER ? FacingMode.ENVIRONMENT : FacingMode.USER
+    )
+  }
+
+  const theme: Theme = useTheme()
+
+  return (
+    <StyledCameraInput width={width}>
+      {webcamPermission !== WebcamPermission.SUCCESS &&
+      !disabled &&
+      !inClearPhotoMode
+        ? AskForCameraPermission({ width })
```

Comment:
nit: I think this line should be written `<AskForCameraPermission width={width} />` to emphasize that it's a react component